src/components/chips/js/chipsDirective.js

```diff
@@ -0,0 +1,159 @@
+	 *   Truncate chip text where possible, but do not truncate entries such that two are indistinguishable.
+	 * Drag and Drop (?)
+	 *   Drag and drop chips between related <md-chips> elements.
+	 *
+	 *
+	 * @param {string=|object=} ng-model A model to bind the list of items to
+	 * @param {string=} placeholder Placeholder text that will be forwarded to the input.
+	 * @param {string=} secondary-placeholder Placeholder text that will be forwarded to the input, displayed when there
+	 *    is at least on item in the list
+	 * @param {boolean=} readonly Disables list manipulation (deleting or adding list items), hiding the input and delete
+	 *    buttons
+	 * @param {expression} md-chip-append An expression expected to convert the input string into an object when adding
+	 *    a chip.
+	 *
+	 * @usage
+	 * <hljs lang="html">
+	 *   <md-chips
+	 *       ng-model="myItems"
+	 *       placeholder="Add an item"
+	 *       readonly="isReadOnly">
+	 *   </md-chips>
+	 * </hljs>
+	 *
+	 */
+
+
+	var MD_CHIPS_TEMPLATE = '\
+			<md-chips-wrap ng-class="{readonly : $mdChipsCtrl.readonly}" class="md-chips">\
+				<div role="presentation">\
+					<div ng-repeat="$chip in $mdChipsCtrl.items"\
+							ng-class="{selected: $mdChipsCtrl.selectedChip == $index}"\
+							ng-click="!$mdChipsCtrl.readonly && $mdChipsCtrl.selectChip($index)"\
+							class="md-chip">\
+					</div>\
+					<div ng-if="!$mdChipsCtrl.readonly" class="md-chip-worker"></div>\
+				</div>\
+			</md-chips-wrap>';
+
+  var CHIP_INPUT_TEMPLATE = '\
+				<input\
+						placeholder="{{$mdChipsCtrl.items.length == 0 ? $mdChipsCtrl.placeholder : $mdChipsCtrl.secondaryPlaceholder}}"\
+						class="md-chip-input"\
+						ng-model="$mdChipsCtrl.chipBuffer"\
+						ng-focus="$mdChipsCtrl.resetSelectedChip()"\
+						ng-keydown="$mdChipsCtrl.defaultInputKeydown($event)">';
+
+  var CHIP_DEFAULT_TEMPLATE = '\
+			<md-chip>\
+				<span>{{$chip}}</span>\
+				<md-button ng-if="!$mdChipsCtrl.readonly"\
+						md-chip-remove>x</md-button>\
+			</md-chip>';
+
+
+  /**
+   * MDChips Directive Definition
+   * @param $mdTheming
+   * @param $log
+   * @ngInject
+   */
+  function MdChips ($mdTheming, $log) {
+    return {
+      template: function(element, attrs) {
+        // Clone the element into an attribute. By prepending the attribute
+        // name with '$', Angular won't write it into the DOM. The cloned
+        // element propagates to the link function via the attrs argument,
+        // where various contained-elements can be consumed.
+        attrs['$mdUserTemplate'] = element.clone();
+        return MD_CHIPS_TEMPLATE;
+      },
+      require: ['ngModel', 'mdChips'],
+      restrict: 'E',
+      controller:   'MdChipsCtrl',
+      controllerAs: '$mdChipsCtrl',
+      bindToController: true,
+      compile: compile,
+      scope: {
+        readonly:             '=readonly',
+        placeholder:          '@',
+        secondaryPlaceholder: '@',
+        mdChipAppend:         '&'
```

Comment:
@typotter 
*  I concur that I do not like the current **mdChipAppend** name. It appears to act like a filter; but I cannot think of a better name right now.
*  For Angular Material, in general, we need to consider which components will dispatch events for which component-internal changes. This prepares use for Angular 2.x... Let's revisit the event debat in a separate issue.